Source/Core/Core/PatchEngine.cpp

```diff
@@ -45,6 +46,53 @@ const char* PatchTypeAsString(PatchType type)
   return s_patch_type_strings.at(static_cast<int>(type));
 }
 
+std::optional<PatchEntry> DeserializeLine(std::string line)
+{
+  std::string::size_type loc = line.find('=');
+  if (loc != std::string::npos)
+    line[loc] = ':';
+
+  const std::vector<std::string> items = SplitString(line, ':');
+  PatchEntry entry;
+
+  if (items.size() < 3)
+    return std::nullopt;
+
+  if (!TryParse(items[0], &entry.address))
+    return std::nullopt;
+  if (!TryParse(items[2], &entry.value))
+    return std::nullopt;
+
+  if (items.size() >= 4)
+  {
+    if (!TryParse(items[3], &entry.comparand))
+      return std::nullopt;
+    entry.conditional = true;
+  }
+
+  const auto iter = std::find(s_patch_type_strings.begin(), s_patch_type_strings.end(), items[1]);
+  const ptrdiff_t type = std::distance(s_patch_type_strings.begin(), iter);
+  if (static_cast<size_t>(type) == s_patch_type_strings.size())
+    return std::nullopt;
+  entry.type = static_cast<PatchType>(type);
```

Comment:
I think it'd be more readable to just check for `iter == s_patch_type_strings.end()` here.

Reply:
Yeah, now that I think about it outside of the context of just trying to translate the old code, I agree.